src/OAuth2/GrantType/AuthorizationCode.php

```diff
@@ -84,6 +84,45 @@ public function validateRequest(RequestInterface $request, ResponseInterface $re
             return false;
         }
 
+        // @TODO: Should we enforce presence of a non-falsy code challenge?
+        if (isset($authCode['code_challenge']) && $authCode['code_challenge']) {
```

Comment:
It's not in the official spec but like other things in OAuth, providing an empty string should be considered the same as omitting the parameter. Maybe more pedantically:

```suggestion
        if (isset($authCode['code_challenge']) && $authCode['code_challenge'] !== '') {
```